### PR TITLE
Standardized CLI options

### LINKED ISSUE #849: [FEATURE] Standardize the CLI option for output directory
### Is your feature request related to a problem? Please describe.
Currently, there are multiple types of "output" used in CLI, which can get confusing.

### Describe the solution you'd like
For anything related to output directory, it should be `outputdir` or something similar.

### Describe alternatives you've considered
N.A.

### Additional context
Update documentation as well.

### PR DESCRIPTION
**Pull request by scap3yvt**
Fixes #849

## Proposed Changes

- updated cli options to standardize them 

## Checklist

- [x] [`CONTRIBUTING`](https://github.com/mlcommons/GaNDLF/blob/master/CONTRIBUTING.md) guide has been followed.
- [x] PR is based on the [current GaNDLF master ](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/keeping-your-local-repository-in-sync-with-github/syncing-your-branch-in-github-desktop?platform=windows).
- [x] Non-breaking change (does **not** break existing functionality): provide **as many** details as possible for _any_ breaking change.
- [x] Function/class source code documentation added/updated (ensure `typing` is used to provide type hints, including and not limited to using `Optional` if a variable has a pre-defined value).
- [x] Code has been [blacked](https://github.com/psf/black#usage) for style consistency and linting.
- [x] If applicable, version information [has been updated in GANDLF/version.py](https://github.com/mlcommons/GaNDLF/blob/master/GANDLF/version.py).
- [x] If adding a git submodule, add to list of exceptions for black styling in [pyproject.toml](https://github.com/mlcommons/GaNDLF/blob/master/pyproject.toml) file.
- [x] [Usage documentation](https://github.com/mlcommons/GaNDLF/blob/master/docs) has been updated, if appropriate.
- [x] Tests added or modified to [cover the changes](https://app.codecov.io/gh/mlcommons/GaNDLF); if coverage is reduced, please give explanation.
- [x] If customized dependency installation is required (i.e., a separate `pip install` step is needed for PR to be functional), please ensure it is reflected in all the files that control the CI, namely: [python-test.yml](https://github.com/mlcommons/GaNDLF/blob/master/.github/workflows/python-test.yml), and all docker files [[1](https://github.com/mlcommons/GaNDLF/blob/master/Dockerfile-CPU),[2](https://github.com/mlcommons/GaNDLF/blob/devcontainer_build_fix/Dockerfile-CUDA11.6),[3](https://github.com/mlcommons/GaNDLF/blob/master/Dockerfile-ROCm)].

### CODE REVIEW

**Comment by sarthakpati** [comment]
@hasan7n @aristizabal95 @VukW - do you have any idea why the MLCube deploy test is failing?

File: gandlf_generateMetrics [hunk 1]
```diff
@@ -56,16 +56,18 @@ if __name__ == "__main__":
...
 
     args = parser.parse_args()
-    assert args.config is not None, "Missing required parameter: config"
-    assert args.inputdata is not None, "Missing required parameter: inputdata"
+
+    # check for required parameters - this is needed here to keep the cli clean
+    for param_none_check in [args.inputDir, args.config]:
+        assert (
+            param_none_check is not None
+        ), f"Missing required parameter: {param_none_check}"
```
Comment (VukW) [inline gandlf_generateMetrics:64]: @sarthakpati the issue is raised here:
```
  File "/GaNDLF/gandlf_generateMetrics", line 61, in <module>
    for param_none_check in [args.inputDir, args.config]:
AttributeError: 'Namespace' object has no attribute 'inputDir'
```
`generateMetrics` doesn't have `inputDir`, but `inputdata` instead

Comment (VukW) [reply gandlf_generateMetrics:64]: Btw why do you check `None`-ness here? both config and inputdata are already marked as `required`:
```
    parser.add_argument(
        "-c",
        "--config",
        ...
        required=True,
        ...
```
So argparse should not allow empty values, isn't it?

**Comment by VukW** [comment]
Hi @sarthakpati! this PR would have a lot of essential conflicts while merging with https://github.com/mlcommons/GaNDLF/pull/845 as all the scripts are refactored heavily while moving to new CLI API. How urgent are these fixes?
- If we want to fix this right now (or, just, before releasing 0.1.0), then may I ask you to fix just a parameters names? It would be much easier for me to solve conflicts later + all the logic around argparse (say, checking if param is required) would be not used soon.
- Or maybe we can release 0.1.0 firstly? And standartize param names later?
- Or, if the question only about param names but not param behavior, we may fix it directly in https://github.com/mlcommons/GaNDLF/pull/845 ?

**Comment by sarthakpati** [comment]
@VukW: I agree - this PR is not critical, and let's change this to get merged into #845 instead, since this is basically only about standardizing param names. Do you think you can help @scap3yvt with this? 

IMO, same goes for #852 - thoughts?

**Comment by VukW** [comment]
@sarthakpati yes, sure, I can help, though not sure how exactly

But it seems to me that it may be easier to start a new branch based on https://github.com/mlcommons/GaNDLF/pull/845 rather than merge conflicts. @scap3yvt what do you think? Can you take #845 as base and add fixes & improvements over it? Or do you need my help for it? If any, feel free to ping me, let's solve this together

**Comment by VukW** [comment]
@sarthakpati concerning #852 - that PR is not meant to be merged, I use it for testing & debugging MLCube CI test that behave in strange way for me

**Comment by sarthakpati** [comment]
This has been superseded by #853

**Pull request closed without merging**